extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -202,6 +202,33 @@ export class AmpLightboxViewer extends AMP.BaseElement {
     this.container_.appendChild(mask);
   }
 
+  /**
+   * Return a cleaned clone of the given element for building
+   * carousel slides with.
+   * @param {!Element} element
+   * @private
+   */
+  cloneLightboxableElement_(element) {
+    const deepClone = !element.classList.contains(
```

Comment:
I think the fact that in case of `<figure lightbox>` the `<amp-img>` is the one gets the lightboxability should be handled in the manger in `processLightboxElement_`, similar to how other exceptions like `amp-carousel` are handled there.  (Note that the carousel handling needs to become recursive now so <figure> in carousel is reprocessed as well)

Reply:
Does it have to be recursive? I.e. do we have to handle figures with carousels in them? Or can we assume that there will be figures in amp-carousels but not vice versa? Aka, do we need to handle carousels inside figures?